lib/routing_constraints.rb

```diff
@@ -35,7 +35,7 @@ def matches?(request)
 
     master = Account.master
     host = request.internal_host
-    master.match_internal_admin_domain?(host) or master.match_internal_domain?(host)
+    master.external_admin_domain == host or master.external_domain == host
```

Comment:
Why this change? I think this should be remained untouched.

Reply:
You're right, I missed that `host` is really `internal_host`.